public/app/plugins/panel/barchart/utils.ts

```diff
@@ -286,7 +289,19 @@ export const preparePlotConfigBuilder: UPlotConfigPrepFn<BarChartOptionsEX> = ({
         tickLabelRotation: vizOrientation.xOri === 1 ? xTickLabelRotation * -1 : 0,
         theme,
         grid: { show: customConfig.axisGridShow },
-      });
+      };
+
+      if (customConfig.axisBorderShow) {
+        axisOpts.border = {
+          show: true,
+        };
+      }
+
+      if (customConfig.axisColorMode === AxisColorMode.Series) {
+        axisOpts.color = seriesColor;
+      }
+
+      builder.addAxis(axisOpts);
```

Comment:
I wonder given how broken options have come up now for two different panels (Barchart + [XY Chart](https://github.com/grafana/grafana/pull/75044)) if we should consider doing an audit of all of the panels we own to 1) make sure options actually work as expected, and 2) make sure options make sense (I remember for XY Chart there are some refactors / rethinking of options needed before pushing it out of `beta` state)